src/Symfony/Component/Security/Core/Encoder/ChainPasswordEncoder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encoder;
+
+/**
+ * Hashes passwords using the best available encoder.
+ * Validates them using a chain of encoders.
+ *
+ * /!\ Don't put a PlaintextPasswordEncoder in the list as that'd mean a leaked hash
```

Comment:
Instead of (or: in addition to) this warning, why not put a check in the constructor that throws an exception when the `PlaintextPasswordEncoder` is passed?

Reply:
I wouldn't do this: technically one might have a reason to do so, we shouldn't close the door.

Reply:
fwiw, I use plaintextpasswordencoder in my test harness.